static/app/views/dashboards/widgets/timeSeriesWidget/plottables/continuousTimeSeries.tsx

```diff
@@ -79,6 +80,15 @@ export abstract class ContinuousTimeSeries<
     return this.#timestamps.at(-1) ?? null;
   }
 
+  clone(cb?: (timeSeries: TimeSeries) => TimeSeries) {
+    const newTimeSeries =
+      typeof cb === 'function' ? cb(this.timeSeries) : cloneDeep(this.timeSeries);
+    return new (this.constructor as new (
+      timeSeries: TimeSeries,
+      config?: TConfig
+    ) => this)(newTimeSeries, this.config);
```

Comment:
This is kinda gross, but don't know a better way. ~Reading~ Skimming through https://github.com/microsoft/TypeScript/issues/3841 has not helped.

I don't think this is truly typesafe? e.g. if `Line` has a diff constructor signature

Reply:
str8 2 jail!!

Yeah, the constructors are different, I think you should implement `.clone()` (or `.slice` below) per implementation of the plottable. One for `Line`, one for `Area`, etc. You can add a helper method to `ContinuousTimeSeries` to share code if there's code to be shared